app/helpers/application_helper.rb

```diff
@@ -5,4 +5,12 @@ def gravatar_for(user, option = { size: 200 })
     hash = Digest::MD5.hexdigest(email_address)
     image_tag("https://www.gravatar.com/avatar/#{hash}?s=#{size}", class: 'rounded mx-auto d-block')
   end
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
```

Comment:
như này thì `||=` vô ích với trường hợp chưa đăng nhập. mỗi lần gọi đều query DB.

```suggestion
    return if session[:user_id].blank?

    @current_user ||= User.find(session[:user_id]) 
```

à mà ông biết tác dụng của cái `||=` chưa đấy?

Reply:
Cái `||=` là nếu @current_user falsy thì sẽ lấy thằng đằng sau đúng hông?
Viết kiểu kia thì nó vẫn sẽ check nếu có session[:user_id] thì với query DB mà nhỉ
Giống cái này https://replit.com/@DinhPham/RegularBasicMapping#main.rb